openapi.yaml

```diff
@@ -1,279 +1,678 @@
+  responses:
+    '400':
+      description: Bad request
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorResult'
+    '404':
+      description: Not found
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorResult'
+    '409':
+      description: Conflict
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorResult'
+    '500':
+      description: Internal server error
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorResult'
+  schemas:
+    LocationResult:
+      properties:
+        links:
+          $ref: '#/components/schemas/Links'
+        data:
+          $ref: '#/components/schemas/LocationsResourceObject'
+        included:
+          type: array
+          items:
+            $ref: '#/components/schemas/ServicesResourceObject'
+    LocationSetResult:
+      properties:
+        links:
+          $ref: '#/components/schemas/Links'
+        data:
+          type: array
+          items:
+            $ref: '#/components/schemas/LocationsResourceObject'
+        included:
+          type: array
+          items:
+            $ref: '#/components/schemas/ServicesIncluded'
+    ServicesResultObject:
+      properties:
+        links:
+          $ref: '#/components/schemas/Links'
+        data:
+          $ref: '#/components/schemas/ServicesResourceObject'
+    ServicesIncluded:
+      properties:
+        id:
+          type: string
+        type:
+          type: string
+        attributes:
+          $ref: '#/components/schemas/ServicesAttributes'
+        links:
+          $ref: '#/components/schemas/ServiceLinks'
+    LocationsResourceObject:
+      properties:
+        id:
+          type: string
+        type:
+          type: string
+        attributes:
+          $ref: '#/components/schemas/Attributes'
+        links:
+          $ref: '#/components/schemas/SelfLink'
+        relationships:
+          $ref: '#/components/schemas/RelationshipsInLocations'
+    ServicesResultObjects:
+      properties:
+        links:
+          $ref: '#/components/schemas/Links'
+        data:
+          type: array
+          items:
+            $ref: '#/components/schemas/ServicesResourceObject'
+    ServicesResourceObject:
+      properties:
+        id:
+          type: string
+        type:
+          type: string
+        attributes:
+          $ref: '#/components/schemas/ServicesAttributes'
+        links:
+          $ref: '#/components/schemas/ServiceLinks'
+        relationships:
+          $ref: '#/components/schemas/RelationshipsInServices'
+    SelfLink:
+      properties:
+        self:
+          type: string
+          description: Url to this resource / location.
+    Attributes:
+      properties:
+        name:
+          type: string
+          description: Name of the location.
```

Comment:
Can you add `description`, `example`, `format` (if applied), and `nullable` fields for all of these attributes? Example: https://github.com/osu-mist/persons-api/blob/v2/openapi.yaml#L891-L920

Reply:
I'm not sure how to determine which fields are nullable -- is there a good source of truth for that information?

Reply:
Not really. I did a lot `select _ from TABLE if _ is null`

Reply:
Hmm, @tsoliangwu0130 do you know?

Reply:
There's no good way to tell for this API. The only way we can tell is to look at the source file I sent you or from AWS ES instance.